goagen/codegen/validation.go

```diff
@@ -170,6 +175,52 @@ func (v *Validator) recurse(att *design.AttributeDefinition, nonzero, required,
 			}
 			buf.WriteString(validation)
 		}
+	} else if h := att.Type.ToHash(); h != nil {
+		// Perform any validation on the hash type such as MinLength, MaxLength, etc.
+		validation := ValidationChecker(att, nonzero, required, hasDefault, target, context, depth, private)
+		first := true
+		if validation != "" {
+			buf.WriteString(validation)
+			first = false
+		}
+		val := v.Code(h.KeyType, true, false, false, "k", context+"[*]", depth+1, false)
+		if val != "" {
+			keyVal := val
+			switch h.KeyType.Type.(type) {
+			case *design.UserTypeDefinition, *design.MediaTypeDefinition:
+				// For user and media types, call the Validate method
+				keyVal = RunTemplate(v.userValT, map[string]interface{}{
+					"depth":  depth + 2,
+					"target": "k",
+				})
+				keyVal = fmt.Sprintf("%sif e != nil {\n%s\n%s}", Tabs(depth+1), val, Tabs(depth+1))
+			}
+			elemVal := v.Code(h.ElemType, true, false, false, "e", context+"[*]", depth+1, false)
```

Comment:
Shouldn't that be outside of the `if val != "" {` ? There may be validations that apply to the elements even if no validation applies to the keys. It seems the code should compute validations for the keys and values separately and render both separately.